Objects/bytearrayobject.c

```diff
@@ -1801,7 +1937,88 @@ bytearray_insert_impl(PyByteArrayObject *self, Py_ssize_t index, int item)
     Py_RETURN_NONE;
 }
 
+static PyObject *
```

Comment:
You can use arg clinic here as well, to avoid manual critical section here.

Reply:
I didn't want to duplicate the docstrings from the respective target functions (is there a way to tell clinic to get it from there). Other option is I can halfway it:
```c
/*[clinic input]
@critical_section
bytearray.isalnum
[clinic start generated code]*/

static PyObject *
bytearray_isalnum_impl(PyByteArrayObject *self)
/*[clinic end generated code: output=1d73eca8a7de07a1 input=1a95492494d02017]*/
{
    return stringlib_isalnum((PyObject *)self, NULL);
}


static PyMethodDef bytearray_methods[] = {
    ...
    {"isalnum", bytearray_isalnum, METH_NOARGS, _Py_isalnum__doc__},
```
Or just duplicate it, your call:
1. Leave as is.
2. Halfway.
3. Duplicate docstring.